src/mlpack/methods/bayesian_ridge/bayesian_ridge.cpp

```diff
@@ -0,0 +1,299 @@
+/**
+ * @file bayesian_ridge.cpp
+ * @author Clement Mercier 
+ *
+ * Implementation of Bayesian Ridge regression.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include "bayesian_ridge.hpp"
+#include <mlpack/core/util/log.hpp>
+#include <mlpack/core/util/timers.hpp>
+
+using namespace mlpack;
+using namespace mlpack::regression;
+
+
+BayesianRidge::BayesianRidge(const bool fitIntercept,
+                             const bool normalize) :
+  fitIntercept(fitIntercept),
+  normalize(normalize)
+{/* Nothing to do */}
+
+float BayesianRidge::Train(const arma::mat& data,
+                           const arma::rowvec& responses)
+{
+  Timer::Start("bayesian_ridge_regression");
+
+  arma::mat phi;
+  arma::rowvec t;
+  arma::colvec vecphitT;
+  arma::mat phiphiT;
+  arma::colvec eigval;
+  arma::mat eigvec;
+  arma::colvec eigvali;
+
+  // Preprocess the data. Center and normalize.
+  CenterNormalize(data,
```

Comment:
Do you think we could add a parameter to turn data scaling on/off? Like we did here:

https://github.com/mlpack/mlpack/blob/594713a25c048abf03b9605648206e879a2b3b2d/src/mlpack/methods/pca/pca.hpp#L42

Reply:
I guess that it is already done by the  the const bool normalize parameter. If it is the case I will rename it scaleData. 
Regarding the fit Intercept parameter, do you think that we could name it centerData instead ?

Reply:
Yes, I think that works.